lib/internal/main/eval_stdin.js

```diff
@@ -24,7 +24,7 @@ readStdin((code) => {
 
   const print = getOptionValue('--print');
   if (getOptionValue('--input-type') === 'module')
-    evalModule(code, print);
+    evalModule(code, print, 'node:stdin');
```

Comment:
CJS is using `[stdin]` instead of `node:stdin` for this, should we chose `node:[stdin]` instead? I don't feel strongly about this at all, but if there are no drawbacks maybe we should aim for consistency.

On another note, I've noticed that this PR makes the following work:

```console
$ echo 'export {test} from "node:stdin"' > test.mjs
$ out/Release/node --input-type=module <<EOF
export let test = 4;
import { test as test2 } from './test.mjs';
console.log({test,test2})
test++;
console.log({test,test2});
EOF
{ test: 4, test2: 4 }
{ test: 5, test2: 5 }
```

While it doesn't seem to work at all on `master`:

```console
$ echo 'export {test} from "./[eval1]"' > test.mjs
$ node --input-type=module <<EOF
export let test = 4;
import { test as test2 } from './test.mjs';
console.log({test,test2})
test++;
console.log({test,test2});
EOF
node:internal/errors:465
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '…/[eval1]' imported from …/test.mjs
    at new NodeError (node:internal/errors:372:5)
    at finalizeResolution (node:internal/modules/esm/resolve:404:11)
    at moduleResolve (node:internal/modules/esm/resolve:963:10)
    at defaultResolve (node:internal/modules/esm/resolve:1172:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:80:40)
    at link (node:internal/modules/esm/module_job:78:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v17.7.2
```

Reply:
wrapping both in `[]` seems good for consistency, then we could also use `[eval]` like CJS.

I'm fine removing the relative importing for these modules as long as it gives a little more consistency with CJS where you can use `require()` to get relative files from `[stdin]`.